src/main/scala/cromwell/engine/backend/BackendCall.scala

```diff
@@ -54,21 +54,37 @@ trait ExecutionHandle {
   def result: ExecutionResult
 }
 
+sealed trait ReturnCode {
+  def code: Int
+
+  def asScriptReturnCode = this match {
+    case script: ScriptReturnCode => Option(code)
+    case _ => None
+  }
+
+  def asBackendReturnCode = this match {
+    case script: BackendReturnCode => Option(code)
+    case _ => None
+  }
```

Comment:
I actually would prefer they didn't implement the same trait at all, and functions make their mind up whether they return one or the other (or both)